Build failure: fontforge with Python 3.9

### Steps To Reproduce

Steps to reproduce the behavior:
1. build nixos from flake on with the current unstable commit with steam enabled (693bc46d169f5af9c992095736e82c3488bf7dbb)

Sorry I can't figure out a better way to pair this down since the standard nix-build command for this seems to still work fine.
### Build log

full log https://gist.github.com/Unverified/4768507e5566b4c5a30452d83119bda1
```
error: builder for '/nix/store/iqwjn60gnyyvn5z1hpq0xcb13rbj1977-fontforge-20230101.drv' failed with exit code 1;
       last 10 log lines:
       > -- Could NOT find Git (missing: GIT_EXECUTABLE)
       > Traceback (most recent call last):
       >   File "<string>", line 1, in <module>
       > AttributeError: module 'sysconfig' has no attribute 'get_preferred_scheme'
       > CMake Error at pyhook/CMakeLists.txt:29 (message):
       >   Could not determine the path to install pyhook binaries to, got '' with
       >   return value 1
       >
       >
       > -- Configuring incomplete, errors occurred!
       For full logs, run 'nix log /nix/store/iqwjn60gnyyvn5z1hpq0xcb13rbj1977-fontforge-20230101.drv'.
error: 1 dependencies of derivation '/nix/store/5dc2kmajlyn228d0m0wkrcwjbhy4h62f-python3-3.9.19-env.drv' failed to build
error: 1 dependencies of derivation '/nix/store/2dn1yqaac4xk43mqi2frjsv0k9h33hcq-monoid-2020-10-26.drv' failed to build
error: 1 dependencies of derivation '/nix/store/20kpyfm414a4van7islg39x68v392k3x-fc-00-nixos-cache.conf.drv' failed to build
error: 1 dependencies of derivation '/nix/store/3rmlwci4w7y4srk709nym8qypi5rxndv-steam-run-usr-target.drv' failed to build
error: 1 dependencies of derivation '/nix/store/jhw0bjkblmqg6wm9h0fkh5bzbaf32bkd-steam-usr-target.drv' failed to build
error: 1 dependencies of derivation '/nix/store/8k6ypcw3z67f21fhvnrmdfk167mm6yvs-xserver.conf.drv' failed to build
error: 1 dependencies of derivation '/nix/store/01bl6hsw6m0dpd0qdw9gd8q3v9pngxys-fontconfig-conf.drv' failed to build
error: 1 dependencies of derivation '/nix/store/04msx087cvfnbnzv4kwnrgcqcw7rkjj9-steam-fhs.drv' failed to build
error: 1 dependencies of derivation '/nix/store/n66cpf58b2zg581qpghwm9rs301s87yc-steam-run-fhs.drv' failed to build
error: 1 dependencies of derivation '/nix/store/fajqnkbwjq6rw8in6ms24ll672kmd46i-unit-display-manager.service.drv' failed to build
error: 1 dependencies of derivation '/nix/store/x8ach8rkbnkmfn020qk8m6yd7m72xg0k-fontconfig-etc.drv' failed to build
error: 1 dependencies of derivation '/nix/store/0wzkj8lvvrgp43ysb2gsaa1i6whzv2pk-steam-bwrap.drv' failed to build
error: 1 dependencies of derivation '/nix/store/xkslmd1z4b8f4d1ggslsy712isr261kg-steam-run-bwrap.drv' failed to build
error: 1 dependencies of derivation '/nix/store/0nm3ils0mdzp247iaxzyrk2rfgs70kmr-etc.drv' failed to build
error: 1 dependencies of derivation '/nix/store/pxswq3kj8px9g8sv1ldm4k9npz67aj0w-steam-run.drv' failed to build
error: 1 dependencies of derivation '/nix/store/a01nmcdcqjhbwk2rkxv2cn9gyx8q4hdh-steam.drv' failed to build
error: 1 dependencies of derivation '/nix/store/xpmpdhnwzrr1c5bpnaa3xi0rxcvf5sdg-nixos-system-Koharu-24.11.20240714.693bc46.drv' failed to build
```

### Additional context

The previous head of unstable that I can find (7e7c39ea35c5cdd002cd4588b03a3fb9ece6fad9) has this failure happen as well. Previous commit 9f4128e00b0ae8ec65918efeba59db998750ead6 seems to be the most recent one where my systems seems to continue to build.

### Notify maintainers

@erictapen
@mweinelt 

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.36, NixOS, 24.11 (Vicuna), 24.11.20240703.9f4128e`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.4`
 - nixpkgs: `/nix/store/dk2rpyb6ndvfbf19bkb2plcz5y3k8i5v-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/data/fonts/monoid/default.nix
+++ b/pkgs/data/fonts/monoid/default.nix
@@ -1,4 +1,4 @@
-{ lib, stdenv, fetchFromGitHub, python3 }:
+{ lib, stdenv, fetchFromGitHub, fetchpatch, python3 }:
 
 stdenv.mkDerivation {
   pname = "monoid";
@@ -15,6 +15,14 @@ stdenv.mkDerivation {
     (python3.withPackages (pp: with pp; [
       fontforge
     ]))
+  ];
+
+  patches = [
+    # port to python 3
+    (fetchpatch {
+      url = "https://salsa.debian.org/fonts-team/fonts-monoid/-/raw/master/debian/patches/0002-fontbuilder.py-Fix-FTBFS-with-non-int-bearing-value.patch";
+      hash = "sha256-IG3KXvOFZfi8dhVuKDYo0onl6ruEe24aFHBgSoMKf9c=";
+    })
   ];
 
   buildPhase = ''

--- a/pkgs/data/fonts/monoid/default.nix
+++ b/pkgs/data/fonts/monoid/default.nix
@@ -1,4 +1,4 @@
-{ lib, stdenv, fetchFromGitHub, python39 }:
+{ lib, stdenv, fetchFromGitHub, python3 }:
 
 stdenv.mkDerivation {
   pname = "monoid";
@@ -12,7 +12,7 @@ stdenv.mkDerivation {
   };
 
   nativeBuildInputs = [
-    (python39.withPackages (pp: with pp; [
+    (python3.withPackages (pp: with pp; [
       fontforge
     ]))
   ];

